src/libvirtApi/domain.js

```diff
@@ -224,32 +225,31 @@ export function domainChangeInterfaceSettings({
     return cockpit.spawn(args, options);
 }
 
-export function domainChangeAutostart ({
+export async function domainChangeAutostart ({
     connectionName,
     vmName,
     autostart,
 }) {
-    return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainLookupByName', [vmName], { timeout, type: 's' })
-            .then(domainPath => {
-                const args = ['org.libvirt.Domain', 'Autostart', cockpit.variant('b', autostart)];
-
-                return call(connectionName, domainPath[0], 'org.freedesktop.DBus.Properties', 'Set', args, { timeout, type: 'ssv' });
-            });
+    const [domainPath] = await call(
+        connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainLookupByName',
+        [vmName], { timeout, type: 's' });
+    await call(
+        connectionName, domainPath, 'org.freedesktop.DBus.Properties', 'Set',
+        ['org.libvirt.Domain', 'Autostart', cockpit.variant('b', autostart)],
+        { timeout, type: 'ssv' });
```

Comment:
Does `await` return implicitly? Because from a quick demo this does not seem to be the case?

```
function resolveAfter2Seconds() {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve('resolved');
    }, 2000);
  });
}

async function asyncCall() {
  console.log('calling');
  await resolveAfter2Seconds();
}

console.log(asyncCall().then(foo => console.log(foo)));
```

```
> "calling"
> [object Promise]
> undefined
```

Reply:
Note that the point of the original `return call` was *not* to return a value (`Set` doesn't return anything), but return the *promise* of the Set() call. `await` waits until it's done, so it should do the same thing. This is actually true for a lot of the action methods here, the callers don't expect a return value, at most a rejection with an error message.

Reply:
In other words, it would IMHO be confusing to do `return await call()` -- while it would technically work (as the value is `undefined`, and the caller doesn't care), it may give the impression that we care about the value there.

Reply:
Aha, hmm I do thing it a bit confusing to read but maybe that's because it's new.